addons/project_timesheet_holidays/models/project_task.py

```diff
@@ -32,7 +32,7 @@ def _search_is_timeoff_task(self, operator, value):
             ['timesheet_task_ids:array_agg(timesheet_task_id)'],
             [],
         )
-        timeoff_task_ids = leave_type_read_group[0]['timesheet_task_ids'] if leave_type_read_group else []
+        timeoff_task_ids = leave_type_read_group[0]['timesheet_task_ids'] if leave_type_read_group[0]['timesheet_task_ids'] else []
```

Comment:
I think we first have to make sure we have a result with the read_group before checking the `timesheet_task_ids` is defined in the result.
```suggestion
        timeoff_task_ids = leave_type_read_group[0]['timesheet_task_ids'] if leave_type_read_group and leave_type_read_group[0]['timesheet_task_ids'] else []
```

Reply:
Not true in all the cases

It is the following read_group in which we have no value in the list (if no project has an id equals to 5000)
```bash
>>> self.env['project.task'].read_group([('project_id', '=', 5000)], ['project_id', 'ids:array_agg(id)'], ['project_id'])
[]  # result
```

Reply:
that's strange. Here the issue occurs because there is no result but the read_group returns something and I don't see the difference with your example : 
```
>>> leave_type_read_group = self.env['hr.leave.type']._read_group( [('timesheet_task_id', '!=', False)],['timesheet_task_ids:array_agg(timesheet_task_id)'],[],)
>>> leave_type_read_group
[{'id': None, '__count': 0, 'timesheet_task_ids': None}]

```

Reply:
In my example, I have a field in the group by 🙂 

I just pointed out we don't have always a item in the result of a read_group. 🙂 

Indeed, in your case since we have no group by, the read_group will add an item to notice no record is found.

Reply:
But since here you always have a result then we can let as you did.

Reply:
> In my example, I have a field in the group by slightly_smiling_face
> 
> I just pointed out we don't have always a item in the result of a read_group. slightly_smiling_face
> 
> Indeed, in your case since we have no group by, the read_group will add an item to notify no record is found.

Ok, it's clearer to me now. thx :)

Reply:
No problem. 🙂